rte/main.go

```diff
@@ -97,15 +98,19 @@ func main() {
 	}
 
 	if parsedArgs.LocalArgs.ExitOnConfigChanges {
-		cw, err := config.NewWatcher(parsedArgs.LocalArgs.ConfigPath, func() error {
-			klog.Infof("configuration file %q changed: exit", parsedArgs.LocalArgs.ConfigPath)
-			os.Exit(0)
-			return nil
-		})
-		if err != nil {
-			klog.Fatalf("cannot watch the configuration file %q: %v", parsedArgs.LocalArgs.ConfigPath, err)
+		if !parsedArgs.LocalArgs.HasConfigFile {
```

Comment:
What if the ConfigMap is created later on? is it for the operator to handle?

Reply:
this is exactly the point. The operator should react somehow to this event. AFAICT, nothing we can fix at RTE level. Perhaps we would need to change the way configMaps are handled (e.g., throwing ideas: makes them immutable so a change of ConfigMap triggers a deamonset change).

I'm not actually sure this PR is an actual improvement over the current state, because now the only way to learn the `exit-on-config-change` feature is auto-disabled is to look into the logs.